web/src/main/java/org/cbioportal/web/StudyController.java

```diff
@@ -18,19 +29,42 @@
 @Api(tags = "Studies", description = " ")
 public class StudyController {
 
-    @RequestMapping(value = "/studies", method = RequestMethod.GET)
+    @Autowired
+    private StudyService studyService;
+
+    @RequestMapping(value = "/studies", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiOperation("Get all studies")
-    public ResponseEntity<List<? extends CancerStudySummary>> getAllStudies(@RequestParam(defaultValue = "SUMMARY") Projection projection,
-                                                                            @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_SIZE) Integer pageSize,
-                                                                            @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_NUMBER) Integer pageNumber) {
+    public ResponseEntity<List<? extends CancerStudySummary>> getAllStudies(
+            @ApiParam("Level of detail of the response")
+            @RequestParam(defaultValue = "SUMMARY") Projection projection,
+            @ApiParam("Page size of the result list")
+            @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_SIZE) Integer pageSize,
+            @ApiParam("Page number of the result list")
+            @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_NUMBER) Integer pageNumber,
+            @ApiParam("Name of the property that the result list is sorted by")
+            @RequestParam(required = false) StudySortBy sortBy,
+            @ApiParam("Direction of the sort")
+            @RequestParam(defaultValue = "ASC") Direction direction) {
 
-        throw new UnsupportedOperationException();
+        if (projection == Projection.META) {
+            HttpHeaders responseHeaders = new HttpHeaders();
+            responseHeaders.add(HeaderKeyConstants.TOTAL_COUNT, studyService.getMetaStudies().getTotalCount()
```

Comment:
I guess this must go here .. but it would be nice if this logic could be encapsulated since it will be reused a lot (the interception of the META projection query and setting of headers)